src/zimwriterfs/zimcreatorfs.cpp

```diff
@@ -159,20 +159,41 @@ void ZimCreatorFS::visitDirectory(const std::string& path)
   closedir(directory);
 }
 
-void ZimCreatorFS::addArticle(const std::string& path)
+void ZimCreatorFS::addFile(const std::string& path)
 {
-  auto farticle = std::make_shared<FileArticle>(this, path);
-  if (farticle->isInvalid()) {
-    return;
+  auto url = path.substr(directoryPath.size()+1);
+  auto mimetype = getMimeTypeForFile(directoryPath, url);
+  auto ns = getNamespaceForMimeType(mimetype, uniqueNamespace);
+  auto itemPath = ns + "/" + url;
+  auto title = std::string{};
+
+  std::unique_ptr<zim::writer::Item> item;
+  if ( mimetype.find("text/html") != std::string::npos
+    || mimetype.find("text/css") != std::string::npos) {
+    auto content = getFileContent(path);
+
+    if (mimetype.find("text/html") != std::string::npos) {
+      auto redirectUrl = parseAndAdaptHtml(content, title, ns[0], url);
+      if (!redirectUrl.empty()) {
+        // This is a redirect.
+        addRedirection(itemPath, title, redirectUrl);
+        return;
+      }
+    } else {
+      adaptCss(content, ns[0], url);
+    }
+    item.reset(new zim::writer::StringItem(itemPath, mimetype, title, content));
+  } else {
+    item.reset(new zim::writer::FileItem(itemPath, mimetype, title, path));
   }
-  addArticle(farticle);
+  addItem(std::move(item));
 }
 
-void ZimCreatorFS::addArticle(std::shared_ptr<zim::writer::Article> article)
+void ZimCreatorFS::addItem(std::shared_ptr<zim::writer::Item> item)
 {
-  Creator::addArticle(article);
-  for (auto& handler: articleHandlers) {
-      handler->handleArticle(article);
+ Creator::addItem(item);
+ for (auto& handler: itemHandlers) {
+     handler->handleItem(item);
   }
 }
```

Comment:
This change must belong to the next commit "Adapt mimetypecounter." (which renames `articleHandlers` to `itemHandlers` and changes the IHandler interface).